src/Directory.Build.props

```diff
@@ -56,6 +56,7 @@
     <GenerateResourceMSBuildArchitecture>CurrentArchitecture</GenerateResourceMSBuildArchitecture>
     <GenerateResourceMSBuildRuntime>CurrentRuntime</GenerateResourceMSBuildRuntime>
 
+    <EnforceCodeStyleInBuild Condition="'$(MSBuildRuntimeType)' == 'Core'">true</EnforceCodeStyleInBuild>
```

Comment:
This is the crux of the change

Reply:
```suggestion
    <!-- This should be unconditional but helps until https://github.com/dotnet/msbuild/pull/8236 gets us a Roslyn dependency-dll fix. -->
    <EnforceCodeStyleInBuild Condition="'$(MSBuildRuntimeType)' == 'Core'">true</EnforceCodeStyleInBuild>
```

Reply:
OK, linking to https://github.com/dotnet/msbuild/pull/8236 explicitly